xbmc/URL.cpp

```diff
@@ -412,6 +412,11 @@ void CURL::SetProtocolOptions(const std::string& strOptions)
   }
 }
 
+void CURL::SetPayload(const std::string& strPayload)
+{
+  m_strPayload = strPayload;
+}
```

Comment:
@arnova Does a Header belongs to the URL itself? Its already part of the URL structure.
Sure, I can also push it int the URLOptions Variant, Is it more straightforward in your eyes?

If you have an better idea, pls. let me know. I only want to finish it soon because its needed by the new inputstream features. Thanx!